src/controllers/AbstractController.ts

```diff
@@ -164,6 +164,24 @@ export default abstract class AbstractController<T extends AbstractService> {
 		return num;
 	}
 
+	protected parseQueryParamArrayOrThrow(
+		n: string[],
+		errorMessage: string
+	): number[] {
+		const isValidArray = Array.isArray(n);
+
+		if (!isValidArray) {
+			throw new BadRequest(errorMessage);
+		}
```

Comment:
```suggestion
		if (!Array.isArray(n)) {
			throw new BadRequest(errorMessage);
		}
```

…but really I would have expected TS to handle this. You sure you can call this with a non-array of strings?

Reply:
No, what I was referring to is that that the compiler should be able to infer any calls that do not comply with the signature, `string[]`? If it can do that then we can know that it's an array and we don't need to check at all.

Reply:
Since we are taking use input via query param I think it makes sense to do a runtime verification

Reply:
Ahh yes, I see what you are saying. That was a mis-judgement on my part. Yea this is definitely an unnecessary check, and we can remove it. Now that i look back the reason I had this was because the `res.query` has 4 types that it can be returned as. 

`QueryString.ParsedQs | string | string[] | QueryString.ParsedQs[]`

therefore before I was checking to make sure the input was an array, where now its redundant because we check that before we even plug it in. Great catch!